docs/v0.6/en/user-guide/client-libraries/go.md

```diff
@@ -36,129 +37,232 @@ import (
 {template ingester-lib-connect%
 
 ```go
-options := []grpc.DialOption{
-    grpc.WithTransportCredentials(insecure.NewCredentials()),
-}
-// To connect a database that needs authentication, for example, those on Greptime Cloud,
-// `Username` and `Password` are needed when connecting to a database that requires authentication.
-// Leave the two fields empty if connecting a database without authentication.
 cfg := greptime.NewCfg("127.0.0.1").
-    WithDatabase("public").      // change to your real database
-    WithPort(4001).              // default port
-    WithAuth("username", "password").            // `Username` and `Password`
-    WithDialOptions(options...). // specify your gRPC dail options
-    WithCallOptions()            // specify your gRPC call options
+    // change the database name to your database name
+    WithDatabase("public").
+    // default port
+    WithPort(4001).
+    // set authentication information
+    WithAuth("username", "password")
+
+client, _ := client.New(cfg)
+```
+%}
+
+{template low-level-object%
+
+```go
+// Construct the table schema for CPU metrics
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+
+// Add a 'Tag' column for host identifiers
+cpuMetric.AddTagColumn("host", types.STRING)
+// Add a 'Timestamp' column for recording the time of data collection
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+// Add 'Field' columns for user and system CPU usage measurements
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+
+// Insert example data
+// NOTE: The arguments must be in the same order as the columns in the defined schema: host, ts, cpu_user, cpu_sys
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-client, _ := greptime.NewClient(cfg)
 ```
 
 %}
 
+{template create-rows%
 
+```go
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+cpuMetric.AddTagColumn("host", types.STRING)
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-{template row-object%
+memMetric, err := table.New("mem_metric")
+if err != nil {
+    // Handle error appropriately
+}
+memMetric.AddTagColumn("host", types.STRING)
+memMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+memMetric.AddFieldColumn("mem_usage", types.FLOAT)
+err = memMetric.AddRow("127.0.0.1", time.Now(), 112)
+if err != nil {
+    // Handle error appropriately
+}
+```
```

Comment:
I think duplicated code exist in `low-level-object` and `create-rows`. Is it ok to leave the whole runnable sample code in only two blocks of `low-level-object` and `orm-style-object`? Thus will make the blocks simple, keeping the sample codes readable and maintainable. And different language can have subtle difference, which is difficult to keep the same code block for different language.

Reply:
like https://docs.influxdata.com/influxdb/v2/api-guide/client-libraries/go/

- introduce import concept and simple code block above
- supply a complete runnable sample code below

Reply:
We can remove the duplicate code about creating `cpuMetric` object, the `memMetric` will be reserved to show how to write multiple rows to table.
The difference between influxdb's doc and GreptimeDB's doc is, the detail user documentation of influxdb is on [Github repository](https://github.com/influxdata/influxdb-client-go), but in GreptimeDB is on the documentation website.
In the detail user documentation, we will try best to write simple example code, but it is inevitable to write duplicate code to show features.

Reply:
I tried to remove the code of creating the `cpuMetric` object, feeling that it broke the reading experience of creating multiple objects, so kept it in the end.